For ATAC-seq, I recommed some useful tools, papers and scripts.

- [ChIP-seq-analysis](https://github.com/crazyhottommy/ChIP-seq-analysis)
- [ATAC-seq data analysis: from FASTQ to peaks](https://yiweiniu.github.io/blog/2019/03/ATAC-seq-data-analysis-from-FASTQ-to-peaks/)

For ATAC-seq normalization method: 
- [ATAC-seq normalization method can significantly affect differential accessibility analysis and interpretation](https://epigeneticsandchromatin.biomedcentral.com/articles/10.1186/s13072-020-00342-y)

Review paper for ATAC-seq:
- [From reads to insight: a hitchhikerâ€™s guide to ATAC-seq data analysis](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-1929-3#Fig1)

Snakemake pipeline for ATAC-seq:
- [pyflow-ATACseq](https://github.com/crazyhottommy/pyflow-ATACseq)
